src/main/java/com/minecolonies/client/gui/WindowHutFarmer.java

```diff
@@ -1,198 +1,187 @@
+     */
+    private Button buttonNextPage;
+
+    /**
+     * List of fields the building seeds.
+     */
+    private List<FieldView> fields = new ArrayList<>();
+
+    /**
+     * ScrollList with the fields.
+     */
+    private ScrollingList   fieldList;
+
+    /**
+     * Constructor for the window of the farmer
      *
-     * @param ignored   Parameter is ignored, since some actions require a button.
-     *                  This method does not
+     * @param building {@link com.minecolonies.colony.buildings.BuildingFarmer.View}
      */
-    private void melonClicked(Button ignored)
+    public WindowHutFarmer(BuildingFarmer.View building)
     {
-        if (building.melon >= MAX_AMOUNT)
-        {
-            return;
-        }
-        building.melon++;
-        removeOthers(MELON);
+        super(building, Constants.MOD_ID + HUT_FARMER_RESOURCE_SUFFIX);
+        registerButton(BUTTON_PREV_PAGE, this::prevClicked);
+        registerButton(BUTTON_NEXT_PAGE, this::nextClicked);
+        registerButton(TAG_BUTTON_ASSIGNMENT_MODE, this::assignmentModeClicked);
+        registerButton(TAG_BUTTON_ASSIGN, this::assignClicked);
     }
 
-
     /**
-     * Action performed when pumpkin button is clicked
-     *
-     * @param ignored   Parameter is ignored, since some actions require a button.
-     *                  This method does not
+     * Fired when assign has been clicked in the field list.
+     * @param button clicked button.
      */
-    private void pumpkinClicked(Button ignored)
+    private void assignClicked(Button button)
     {
-        if (building.pumpkin >= MAX_AMOUNT)
+        int row = fieldList.getListElementIndexByPane(button);
+        FieldView field = fields.get(row);
+
+        if (button.getLabel().equals(RED_X))
         {
-            return;
+            button.setLabel(APPROVE);
+            MineColonies.getNetwork().sendToServer(new AssignFieldMessage(building, false, field.getId()));
+            fields.get(row).setTaken(false);
+            field.setOwner("");
+        }
+        else
+        {
+            button.setLabel(RED_X);
+            MineColonies.getNetwork().sendToServer(new AssignFieldMessage(building, true, field.getId()));
+            field.setTaken(true);
+            field.setOwner(building.getWorkerName());
         }
-        building.pumpkin++;
-        removeOthers(PUMPKIN);
-    }
 
+        fieldList.refreshElementPanes();
+    }
 
     /**
-     * Action performed when carrot button is clicked
-     *
-     * @param ignored   Parameter is ignored, since some actions require a button.
-     *                  This method does not
+     * Fired when the assignment mode has been toggled.
+     * @param button clicked button.
      */
-    private void carrotClicked(Button ignored)
+    private void assignmentModeClicked(Button button)
     {
-        if (building.carrot >= MAX_AMOUNT)
+        if(button.getLabel().equals(LanguageHandler.format("com.minecolonies.gui.hiring.off")))
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "com.minecolonies.gui.hiring.off" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1192)